src/types/optional_nb.cpp

```diff
@@ -0,0 +1,31 @@
+#include "engine/api/base_parameters.hpp"
+
+#include <nanobind/nanobind.h>
+
+namespace nb = nanobind;
+
+void init_Optional(nb::module_& m) {
+    using OptionalBool = boost::optional<bool>;
```

Comment:
it needs to be boost instead of std? or is osrm c++14 still?

Reply:
it is already c++17, but some boost things are still used there unfortunately...